internal/install/types/discovery_manifest.go

```diff
@@ -29,3 +35,57 @@ type MatchedProcess struct {
 func (d *DiscoveryManifest) AddMatchedProcess(p MatchedProcess) {
 	d.Processes = append(d.Processes, p)
 }
+
+func (d *DiscoveryManifest) ConstrainRecipes(allRecipes []Recipe) []Recipe {
+	var recipes []Recipe
+
+	for _, recipe := range allRecipes {
+		if len(recipe.InstallTargets) == 0 {
+			log.Warnf("recipe has no InstallTargets: %s", recipe.Name)
+		}
+
+		for _, target := range recipe.InstallTargets {
+			if target.KernelArch != "" {
+				if !strings.EqualFold(target.KernelArch, d.KernelArch) {
+					continue
+				}
+			}
+
+			if target.KernelVersion != "" {
+				if !strings.EqualFold(target.KernelVersion, d.KernelVersion) {
+					continue
+				}
+			}
+
+			if target.Os != "" {
+				if !strings.EqualFold(string(target.Os), d.OS) {
+					continue
+				}
+			}
+
+			if target.Platform != "" {
+				if !strings.EqualFold(string(target.Platform), d.Platform) {
+					continue
+				}
+			}
+
+			if target.PlatformFamily != "" {
+				if !strings.EqualFold(string(target.PlatformFamily), d.PlatformFamily) {
+					continue
+				}
+			}
+
+			if target.PlatformVersion != "" {
+				if !strings.EqualFold(target.PlatformVersion, d.PlatformVersion) {
+					continue
+				}
+			}
+
+			recipes = append(recipes, recipe)
+		}
+	}
+
+	log.Debugf("%d embedded recipes found for manifest", len(recipes))
```

Comment:
Nitpick, do we want to say `embedded`?

Reply:
Oh good call.  I think that was a copy-paste from an earlier version.